server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -756,9 +761,22 @@ private void mergeMetadataRecords(MergeInfo info) throws AccumuloException {
         } else if (ServerColumnFamily.DIRECTORY_COLUMN.hasColumns(key)) {
           var allVolumesDir = new AllVolumesDirectory(range.tableId(), value.toString());
           bw.addMutation(manager.getContext().getAmple().createDeleteMutation(allVolumesDir));
+        } else if (HostingColumnFamily.GOAL_COLUMN.hasColumns(key)) {
+          TabletHostingGoal thisGoal = TabletHostingGoalUtil.fromValue(value);
+          goals.add(thisGoal);
         }
       }
 
+      // Set the TabletHostingGoal for this tablet based on the goals of the other tablets in
```

Comment:
Could move all of this logic down past the next loop AND get the hosting goal of the last tablet in the next scan loop.  Then the decision would be based on all tablets instead of every tablet but the last.

Reply:
Updated in d3975ec